pkg/core/blockchain.go

```diff
@@ -303,14 +303,22 @@ func NewBlockchain(s storage.Store, cfg config.ProtocolConfiguration, log *zap.L
 // must be called before `bc.Run()` to avoid data race.
 func (bc *Blockchain) SetOracle(mod native.OracleService) {
 	orc := bc.contracts.Oracle
-	md, ok := orc.GetMethod(manifest.MethodVerify, -1)
-	if !ok {
-		panic(fmt.Errorf("%s method not found", manifest.MethodVerify))
+	if mod != nil {
+		md, ok := orc.GetMethod(manifest.MethodVerify, -1)
+		if !ok {
+			panic(fmt.Errorf("%s method not found", manifest.MethodVerify))
+		}
+		mod.UpdateNativeContract(orc.NEF.Script, orc.GetOracleResponseScript(),
+			orc.Hash, md.MD.Offset)
+		keys, _, err := bc.contracts.Designate.GetDesignatedByRole(bc.dao, noderoles.Oracle, bc.BlockHeight())
```

Comment:
Is it safe to use `bc.dao` without any mutexes?

Reply:
Yes, it's DAO's problem.